syntax = "proto3";
import "google/protobuf/struct.proto";


service WeightsTransfer {
    rpc SendWeights(ClientGrpcMessage) returns (TransferStatus);
}

// Message for the configuration
message LoraConfig {
    string config_name = 1;
    google.protobuf.Value config_value = 2;
}

message ClientGrpcMessage {
  message SendParameters {
    string client_id = 1;
    int64 train_dataset_length = 2;
    map<string, bytes> new_model_weight = 3;
    repeated LoraConfig lora_config = 4;
  }
  message GetNewVersion {
    string version_path = 1;
  }
  oneof msg {
    SendParameters send_parameters = 1;
    GetNewVersion get_new_version = 2;
  }
}

message TransferStatus {
    bool code = 1;
    string message = 2;
}